lib/internal/streams/writable.js

```diff
@@ -537,16 +529,18 @@ function clearBuffer(stream, state) {
     return;
   }
 
+  state.bufferProcessing = true;
```

Comment:
I don't understand why this is better?

Reply:
Unnecessary length operation was done before `state.bufferProcessing` assignment is done. This change eliminates that.

Reply:
I still don't see it. This is actually slower for the case where bufferedLength is 0... since it unnecessarily sets and unsets bufferProcessing.